### ISSUE TITLE
Adding CHART_DIRECTIVES causes app to error

### ISSUE THREAD

**Issue by nanomoffet** [valor-software/ng2-charts#258]
Brand new project using Angular - CLI configured like so:

system-config.ts

```
/** Map relative paths to URLs. */
const map: any = {
'ng2-charts': 'vendor/ng2-charts'
};
/** User packages configuration. */
const packages: any = {
'ng2-charts': {
main: 'bundles/ng2-charts.js'

}

};
```

Angular-cli-build.js

```
vendorNpmFiles: [
    ...
      'chart.js/dist/Chart.min.js',
      'ng2-charts/bundles/*.js'
    ]
```

Component's import and Component code:

```
import { Component } from '@angular/core';
import {CORE_DIRECTIVES, FORM_DIRECTIVES, NgClass} from '@angular/common';
import { CHART_DIRECTIVES } from 'ng2-charts';

@Component({
  moduleId: module.id,
  selector: 'charts2-app',
  templateUrl: 'charts2.component.html',
  styleUrls: ['charts2.component.css'],
directives: [NgClass, CORE_DIRECTIVES, FORM_DIRECTIVES, CHART_DIRECTIVES]
})
```

The moment I add CHART_DIRECTIVES to the Component directives array, my app no longer works and I get the following error stack in the console:

EXCEPTION: Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent'
```
browser_adapter.ts:78EXCEPTION: Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent'BrowserDomAdapter.logError @ browser_adapter.ts:78BrowserDomAdapter.logGroup @ browser_adapter.ts:89ExceptionHandler.call @ exception_handler.ts:53(anonymous function) @ application_ref.ts:341ZoneDelegate.invoke @ zone.js:323NgZoneImpl.inner.inner.fork.onInvoke @ ng_zone_impl.ts:67ZoneDelegate.invoke @ zone.js:322Zone.run @ zone.js:216(anonymous function) @ zone.js:571ZoneDelegate.invokeTask @ zone.js:356NgZoneImpl.inner.inner.fork.onInvokeTask @ ng_zone_impl.ts:56ZoneDelegate.invokeTask @ zone.js:355Zone.runTask @ zone.js:256drainMicroTaskQueue @ zone.js:474ZoneTask.invoke @ zone.js:426
browser_adapter.ts:88EXCEPTION: Error: Uncaught (in promise): Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent'
browser_adapter.ts:78EXCEPTION: Error: Uncaught (in promise): Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent'BrowserDomAdapter.logError @ browser_adapter.ts:78BrowserDomAdapter.logGroup @ browser_adapter.ts:89ExceptionHandler.call @ exception_handler.ts:53(anonymous function) @ application_ref.ts:304schedulerFn @ async.ts:131SafeSubscriber.__tryOrUnsub @ Subscriber.ts:240SafeSubscriber.next @ Subscriber.ts:192Subscriber._next @ Subscriber.ts:133Subscriber.next @ Subscriber.ts:93Subject._finalNext @ Subject.ts:154Subject._next @ Subject.ts:144Subject.next @ Subject.ts:90EventEmitter.emit @ async.ts:117NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.ts:138NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.ts:90ZoneDelegate.handleError @ zone.js:327Zone.runGuarded @ zone.js:233_loop_1 @ zone.js:487drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
browser_adapter.ts:78STACKTRACE:BrowserDomAdapter.logError @ browser_adapter.ts:78ExceptionHandler.call @ exception_handler.ts:56(anonymous function) @ application_ref.ts:304schedulerFn @ async.ts:131SafeSubscriber.__tryOrUnsub @ Subscriber.ts:240SafeSubscriber.next @ Subscriber.ts:192Subscriber._next @ Subscriber.ts:133Subscriber.next @ Subscriber.ts:93Subject._finalNext @ Subject.ts:154Subject._next @ Subject.ts:144Subject.next @ Subject.ts:90EventEmitter.emit @ async.ts:117NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.ts:138NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.ts:90ZoneDelegate.handleError @ zone.js:327Zone.runGuarded @ zone.js:233_loop_1 @ zone.js:487drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
browser_adapter.ts:78Error: Uncaught (in promise): Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent'
    at resolvePromise (zone.js:538)
    at PromiseCompleter.reject (zone.js:515)
    at eval (application_ref.ts:340)
    at ZoneDelegate.invoke (zone.js:323)
    at Object.NgZoneImpl.inner.inner.fork.onInvoke (ng_zone_impl.ts:67)
    at ZoneDelegate.invoke (zone.js:322)
    at Zone.run (zone.js:216)
    at zone.js:571
    at ZoneDelegate.invokeTask (zone.js:356)
    at Object.NgZoneImpl.inner.inner.fork.onInvokeTask (ng_zone_impl.ts:56)BrowserDomAdapter.logError @ browser_adapter.ts:78ExceptionHandler.call @ exception_handler.ts:57(anonymous function) @ application_ref.ts:304schedulerFn @ async.ts:131SafeSubscriber.__tryOrUnsub @ Subscriber.ts:240SafeSubscriber.next @ Subscriber.ts:192Subscriber._next @ Subscriber.ts:133Subscriber.next @ Subscriber.ts:93Subject._finalNext @ Subject.ts:154Subject._next @ Subject.ts:144Subject.next @ Subject.ts:90EventEmitter.emit @ async.ts:117NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.ts:138NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.ts:90ZoneDelegate.handleError @ zone.js:327Zone.runGuarded @ zone.js:233_loop_1 @ zone.js:487drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
zone.js:461 Unhandled Promise rejection: Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent' ; Zone: angular ; Task: Promise.then ; Value: BaseException {message: "Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent'", stack: "Error: Unexpected directive value 'undefined' on t…ocalhost:4200/vendor/zone.js/dist/zone.js:216:44)"}consoleError @ zone.js:461_loop_1 @ zone.js:490drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
zone.js:463 Error: Uncaught (in promise): Unexpected directive value 'undefined' on the View of component 'Charts2AppComponent'(…)consoleError @ zone.js:463_loop_1 @ zone.js:490drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
```

Seems related to https://github.com/valor-software/ng2-charts/issues/223 however angular-cli projects has ts configured to emit metadata by default.

tsconfig.json:

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "noImplicitAny": false,
    "outDir": "../dist/",
    "rootDir": ".",
    "sourceMap": true,
    "target": "es5",
    "inlineSources": true
  },

  "files": [
    "main.ts",
    "typings.d.ts"
  ]
}
```

**Comment by mattblang** [comment]
I'm having the same problem. I didn't use the `CLI` to generate my project, but I am using SystemJS as well. Also have `emitDecoratorMetadata` set to `true`.

**Comment by mattblang** [comment]
@nanomoffet I changed my `SystemJS config` to match what I saw in [this ticket](https://github.com/valor-software/ng2-charts/issues/116). The errors are gone now, though I still don't see a chart (which I am in the process of diagnosing).

**Comment by nanomoffet** [comment]
Well I got it working, but the above solution didn't do it. I actually had to go through and manually add all dependencies on down from ng2-charts as mentioned in this ticket for the angular-cli https://github.com/angular/angular-cli/issues/845#issuecomment-221288257

Perhaps more of a CLI issue than a ng2-charts issue, as the adding of 3rd party libraries seems to be a convoluted endeavor for the CLI in general.

Note that I too had to apply the styles fix even after that system.js setup loaded the modules.
